### ISSUE TITLE
Feature Request: Use San Francisco font instead of Helvetica Neue for iOS

### ISSUE THREAD

**Issue by dylancompanjen** [callstack/react-native-paper#590]
Since iOS9, San Francisco became the default font. The library is using Helvetica Neue though (While San Francisco is available in React Native.
https://material.io/design/platform-guidance/cross-platform-adaptation.html#cross-platform-guidelines

**Comment by ferrannp** [comment]
We should do this but that would be a breaking change. We need to think where and how to adopt it.

**Comment by andrictham** [comment]
@ferrannp 

Right now, if you want to set San Francisco as a font on iOS, you can override the fonts with `System` in your custom theme, like so:

```
import { DefaultTheme, Provider as PaperProvider } from 'react-native-paper';

const ModifiedDefaultTheme = {
  ...DefaultTheme,
  fonts: {
    thin: 'System',
    light: 'System',
    regular: 'System',
    medium: 'System',
  }
}
```

Then do

```
<PaperProvider theme={ModifiedDefaultTheme}>
  <App />
</PaperProvider>
```

The downside of this is, all the fonts will be in the default weight. The reason is, iOS sets font weight using the `fontWeight` style property, but Android doesn’t – it takes in a different `fontFamily` per font weight.

The way theming works will have to be more cross-platform friendly. Is there a way we could somehow use [react-native-typography](https://github.com/hectahertz/react-native-typography) to achieve this?

**Comment by satya164** [comment]
I think we need to change our theme to have these like:

```js
  fonts: {
    regular: { fontFamily: 'System', fontWeight: 'normal' },
    ...
  }
```

However, it'll be a breaking change. For now, the workaround is to bundle those fonts yourself.

**Comment by andrictham** [comment]
@satya164 How would that workaround work, exactly?

**Comment by brunotalhate** [comment]
Following @satya164 comment, the workaround may return a style object with fontFamily and/or fontWeight instead of the fontFamily attribute. Looks like the change lies in change the fontFamily to a font config.

Using the `Button` component as example, the actual style:
```
const fontFamily = theme.fonts.medium
...
<Text
  numberOfLines={1}
  style={[
    styles.label,
    compact && styles.compactLabel,
    textStyle,
    { fontFamily },
  ]}
>
...
```

May become:
```
const fontStyle = theme.fonts.medium
...
<Text
  numberOfLines={1}
  style={[
    styles.label,
    compact && styles.compactLabel,
    textStyle,
    fontStyle,
  ]}
>
```

**Comment by ferrannp** [comment]
Fixed in 3.0.
